lib/elixir/src/elixir_erl_for.erl

```diff
@@ -92,37 +109,36 @@ collect_filters([H | T], Acc) ->
 collect_filters([], Acc) ->
   {Acc, []}.
 
-build_inline(Ann, Clauses, Expr, Into, _Var, Acc, S) ->
+build_inline(Ann, Clauses, Expr, Uniq, Into, _Var, IntoAcc, UniqAcc, S) ->
   case lists:all(fun(Clause) -> element(1, Clause) == bin end, Clauses) of
     true  -> build_comprehension(Ann, Clauses, Expr, Into);
```

Comment:
You need to make sure that we also don't enter this clause if Uniq has been set to true. You can test this by using only binary generators and making sure that uniq: true also works there.